src/main/java/org/mule/extension/validation/api/error/BasicValidationErrorType.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.validation.api.error;
+
+import static java.util.Collections.singleton;
+import static org.mule.extension.validation.api.error.ValidationErrorType.VALIDATION;
+import org.mule.extension.validation.api.ValidationExtension;
+import org.mule.runtime.extension.api.annotation.error.ErrorTypeProvider;
+import org.mule.runtime.extension.api.error.ErrorTypeDefinition;
+
+import java.util.Set;
+
+/**
+ * General {@link ErrorTypeDefinition} for {@link ValidationExtension} operations.
+ *
+ * @since 1.0
+ */
+public class BasicValidationErrorType implements ErrorTypeProvider {
```

Comment:
It provides the most general validation error, to be used on custom validators, for example.